Tests/GitHubContents.tests.ps1

```diff
@@ -0,0 +1,151 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+<#
+.Synopsis
+   Tests for GitHubContents.ps1 module
+#>
+
+# This is common test code setup logic for all Pester test files
+$moduleRootPath = Split-Path -Path $PSScriptRoot -Parent
+. (Join-Path -Path $moduleRootPath -ChildPath 'Tests\Common.ps1')
+
+try {
+    # Define Script-scoped, readonly, hidden variables.
+    @{
+        repoGuid       = [Guid]::NewGuid().Guid
+        readmeFileName = "README.md"
+    }.GetEnumerator() | ForEach-Object {
+        Set-Variable -Force -Scope Script -Option ReadOnly -Visibility Private -Name $_.Key -Value $_.Value
+    }
+
+    # Need this twice since we need to use the above var here
```

Comment:
Need what twice?

Reply:
The reason why there are 2x blocks defining "Script-scoped, readonly, hidden variables" is because the 2nd block requires a variable from the first block and it's not possible to use a variable within another variable declared in the same block. This syntax is new to me (sorry i'm not powershell wizard... yet ;) so I'm unsure what the best practice is here. Instead I can define the `repoGuid` as a normal variable and then use that within these script-scoped variables? Just let me know what the best practice is in this case. Thanks!

Reply:
ok, I've provided a an alternate way of writing that comment to make things a bit more clear.